### DISCUSSION TITLE
Difference in Timestamp Values between spark and duckdb

### DISCUSSION THREAD

**Discussion by praveenkumarb1207** [duckdb/duckdb#13327] [Q&A]
I created a Parquet file using a DataFrame in PySpark, which includes a timestamp column. After writing the Parquet file, I read it using both Spark and DuckDB. I noticed a discrepancy in the timestamp column values between the two DataFrames. 

**Code to Reproduce the Issue**

The following code will create a Parquet file and read it using Spark and DuckDB, storing the results in their respective DataFrames. 

```
from pyspark.sql import SparkSession
import duckdb
import pandas as pd
from pyspark.sql import SparkSession
 
# Initialize Spark session
spark = SparkSession.builder.appName("Test").getOrCreate()

# Creating a parquet file 
data = Seq(
      ("2024-08-06 12:00:00", "UTC"),
      ("2024-08-06 12:00:00", "Europe/Paris"),
      ("2024-08-06 12:00:00", "America/New_York"),
      ("2024-08-06 12:00:00", "Asia/Tokyo"),
      ("2024-08-06 12:00:00", "Australia/Sydney")
    ).toDF("timestamp", "timezone")

// Convert the timestamp string to a TimestampType
df = data.withColumn("timestamp", to_timestamp(col("timestamp")))

df.write.mode("overwrite").save("file:///home/user/spark-timestamp.parquet")



# Code to read the parquet file using Spark

pyspark_df = spark.read.parquet("file:///home/user/spark-timestamp.parquet")


# Establish a DuckDB connection 

conn = duckdb.connect() 

# Code to read the parquet file using DuckDb 

duckdb_df = conn.execute("SELECT * FROM '/home/user/spark-timestamp.parquet/**/*.parquet'").fetchdf()
```

print the data from the Spark DataFrame.

```
spark_df.show() 
```

Result : 

```
+-------------------+----------------+
|          timestamp|        timezone|
+-------------------+----------------+
|2024-08-06 12:00:00|Australia/Sydney|
|2024-08-06 12:00:00|America/New_York|
|2024-08-06 12:00:00|    Europe/Paris|
|2024-08-06 12:00:00|      Asia/Tokyo|
|2024-08-06 12:00:00|             UTC|
+-------------------+----------------+

```

 print the data from the DuckDb Dataframe.

```
print(duckdb_df)
```
Result : 

```
      timestamp                    timezone
0 2024-08-06 06:30:00               UTC
1 2024-08-06 06:30:00      Europe/Paris
2 2024-08-06 06:30:00  America/New_York
3 2024-08-06 06:30:00        Asia/Tokyo
4 2024-08-06 06:30:00  Australia/Sydney

```

We can observe a discrepancy in the timestamp values, with a shift of -5:30 in the values. Despite the differences between the DataFrames, is there a way to make these values consistent between Spark and DuckDB?Additionally, it was observed that when a Parquet file created by DuckDB is read using Spark, Spark adds 5:30 to the timestamp values. There is no discrepancy when the same Parquet file is read using DuckDB.

**Comment by soerenwolfers** [comment]
Run

```
SET TIMEZONE='UTC'
```

first thing in duckdb. 


if you want to know more, read https://duckdb.org/2022/01/06/time-zones.html
